parser/parser/test_taxonomy_parser.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+import taxonomy_parser as parser
+import re
+
+
+def test_normalized_filename():
+    x = parser.Parser()
+    normalizer = x.normalized_filename
+    name = normalizer("test")
+    assert name == "test.txt"
+    name = normalizer("test.txt")
+    assert name == "test.txt"
+    name = normalizer("t")
+    assert name == "t.txt"
+
+def test_fileiter():
+    x=parser.Parser()
+    file=x.file_iter("test.txt")
+    counter=0
+    language_code_prefix = re.compile('[a-zA-Z][a-zA-Z]:')
+    for _,line in file:
+        counter+=1
+        assert line == '' or line[0]=="#" or ":" in line
+        if counter==27:
+            assert line == "carbon_footprint_fr_foodges_value:fr:10"
+    assert counter == 37+1
+
+def test_normalizing():
+    x=parser.Parser()
+    text="Numéro #1, n°1 des ¾ des Français*"
+    text=x.normalizing(text,"fr")
+    assert text == "numero-1-n-1-des-des-francais"
+    text="Randôm Languäge wìth àccénts"
+    text=x.normalizing(text,"fr")
+    assert text == "random-language-with-accents"
+
+def test_create_nodes():
```

Comment:
You should put integration test in a different file. (even better create tests/unit and tests/integration and put each kind of file in its directory).

It's always better to distinguish them.

You should add a session fixture to remove all entries at startup (using a Parser instance if you want, or creating a session)

see https://docs.pytest.org/en/7.1.x/how-to/fixtures.html